src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -66,11 +66,12 @@ export const ClientSideStateUpdate = ({
 
     if (viewSettings) {
       store.dispatch(setViewSettings(viewSettings))
-      viewSettings.filterOptions?.type == FilterOptionsKeywords.CLIENTS
+      const view = viewSettingsTemp ? viewSettingsTemp.filterOptions : viewSettings.filterOptions
+      view?.type == FilterOptionsKeywords.CLIENTS
         ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.CLIENT }))
-        : viewSettings.filterOptions?.type == FilterOptionsKeywords.TEAM
+        : view?.type == FilterOptionsKeywords.TEAM
           ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.IUS }))
-          : viewSettings.filterOptions?.type == ''
+          : view.type == ''
             ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
             : store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
```

Comment:
Why is this tertiary operator going down 4 whole levels??

You can fetch the dispatch function using an object lookup for view.type